source/utilities/formatters/test.js

```diff
@@ -76,5 +76,18 @@ describe("Testing formatters", () => {
       expect(separateThousands("-1000.1234567")).toEqual("-1,000.1234567");
       expect(separateThousands("-10000.1234567")).toEqual("-10,000.1234567");
     });
-  })
+  });
+
+  describe("snakeToCamelCase", () => {
+    const { snakeToCamelCase } = formatters;
+    it("format snake case string into camel case", () => {
+      expect(snakeToCamelCase("snake_case_text")).toEqual("snakeCaseText");
+      expect(snakeToCamelCase("snakeCase_text")).toEqual("snakecaseText");
+      expect(snakeToCamelCase("snake_caseText")).toEqual("snakeCasetext");
+    });
+    it("format snake case string into camel case including the first letter", () => {
+      expect(snakeToCamelCase("snake_case_text", true)).toEqual("SnakeCaseText");
+    });
```

Comment:
Could you add some other test cases? For example

- [ ] Throw an exception if the intended text starts with invalid (invalid to use as variable) characters as *^%-... or numbers
- [ ] Throw an exception if the intended text contains in between invalid characters (invalid to use as variable) such as *^%-({...

You can base the development of this functionality with [this article](https://mathiasbynens.be/notes/javascript-identifiers) and test it with [this site](https://mothereff.in/js-variables#yea)
Regex will be your friend.